File: src/type/Alert.ts
// ── Types ────────────────────────────────────────────────────────────────────

export type AlertStatus =
  | 'pending'
  | 'verified'
  | 'escalated'
  | 'en_route'
  | 'resolved'
  | 'dismissed';

export type AlertSource = 'ai' | 'passenger';

export interface Alert {
  id: string;
  trainId: number | string;
  coachId: number | string;
  coach?: number | string | null;
  door?: string;
  line: string;
  lineId: string;
  station: string;
  platform?: string;
  
  time: string;
  date: string;
  datetime?: string;
  elapsed: string | number;

  status: AlertStatus;
  source: AlertSource;
  type: string;

  confidence?: number | null;
  deviceId?: string | null;

  reportedBy?: string;
  passengerComment?: string;
  imageUrl?: string;
  snapshotUrl?: string;

  // Audit trail
  verifiedBy?: string | null | undefined;
  verifiedAt?: string | null | undefined;
  verifiedComment?: string | null | undefined;
  escalatedBy?: string | null | undefined;
  escalatedAt?: string | null | undefined;
  escalatedComment?: string | null | undefined;
  enrouteBy?: string | null | undefined;
  enrouteAt?: string | null | undefined;
  enrouteComment?: string | null | undefined;
  resolvedBy?: string | null | undefined;
  resolvedAt?: string | null | undefined;
  resolvedComment?: string | null | undefined;
  dismissedBy?: string | null | undefined;
  dismissedAt?: string | null | undefined;
  dismissedComment?: string | null | undefined;
}

File: backend/Models/DTOs/Alert.cs
namespace backend.Models.DTOs
{
    public class AlertDTO
{
    public string Id { get; set; } = null!;

    public int? TrainId { get; set; }
    public int? CoachId { get; set; }

    public string Line { get; set; } = null!;
    public string LineId { get; set; } = null!;
    public string Station { get; set; } = null!;

    public string Status { get; set; } = null!;
    public string Source { get; set; } = null!;

    public string Time { get; set; } = null!;
    public string Date { get; set; } = null!;
    public string Datetime { get; set; } = null!;

    public int Elapsed { get; set; }

    public decimal? Confidence { get; set; }
    public string? DeviceId { get; set; }
    public string? ImageUrl { get; set; }

    // Reporter
    public string? ReportedBy { get; set; }
    public string? PassengerComment { get; set; }

    // Status timeline
    public string? VerifiedBy { get; set; }
    public string? VerifiedAt { get; set; }
    public string? VerifiedComment { get; set; }

    public string? EnrouteBy { get; set; }
    public string? EnrouteAt { get; set; }
    public string? EnrouteComment { get; set; }

    public string? ResolvedBy { get; set; }
    public string? ResolvedAt { get; set; }
    public string? ResolvedComment { get; set; }

    public string? EscalatedBy { get; set; }
    public string? EscalatedAt { get; set; }
    public string? EscalatedComment { get; set; }

    public string? DismissedBy { get; set; }
    public string? DismissedAt { get; set; }
    public string? DismissedComment { get; set; }
}
}